packages/@aws-cdk/aws-appmesh/lib/header-match.ts

```diff
@@ -0,0 +1,167 @@
+   * @param headerValue The exact value to test against
+   */
+  static valueIsNot(headerName: string, headerValue: string): HeaderMatch {
+    return new HeaderMatchImpl(headerName, true, { exact: headerValue });
+  }
+
+  /**
+   * The value of the header with the given name in the request must start with
+   * the specified characters.
+   *
+   * @param headerName the name of the header to match against
+   * @param prefix The prefix to test against
+   */
+  static valueStartsWith(headerName: string, prefix: string): HeaderMatch {
+    return new HeaderMatchImpl(headerName, false, { prefix });
+  }
+
+  /**
+   * The value of the header with the given name in the request must not start
+   * with the specified characters.
+   *
+   * @param headerName the name of the header to match against
+   * @param prefix The prefix to test against
+   */
+  static valueDoesNotStartWith(headerName: string, prefix: string): HeaderMatch {
+    return new HeaderMatchImpl(headerName, true, { prefix });
+  }
+
+  /**
+   * The value of the header with the given name in the request must end with
+   * the specified characters.
+   *
+   * @param headerName the name of the header to match against
+   * @param suffix The suffix to test against
+   */
+  static valueEndsWith(headerName: string, suffix: string): HeaderMatch {
+    return new HeaderMatchImpl(headerName, false, { suffix });
+  }
+
+  /**
+   * The value of the header with the given name in the request must not end
+   * with the specified characters.
+   *
+   * @param headerName the name of the header to match against
+   * @param suffix The suffix to test against
+   */
+  static valueDoesNotEndWith(headerName: string, suffix: string): HeaderMatch {
+    return new HeaderMatchImpl(headerName, true, { suffix });
+  }
+
+  /**
+   * The value of the header with the given name in the request must include
+   * the specified characters.
+   *
+   * @param headerName the name of the header to match against
+   * @param regex The regex to test against
+   */
+  static valueMatchesRegex(headerName: string, regex: string): HeaderMatch {
+    return new HeaderMatchImpl(headerName, false, { regex });
+  }
+
+  /**
+   * The value of the header with the given name in the request must not
+   * include the specified characters.
+   *
+   * @param headerName the name of the header to match against
+   * @param regex The regex to test against
+   */
+  static valueDoesNotMatchRegex(headerName: string, regex: string): HeaderMatch {
+    return new HeaderMatchImpl(headerName, true, { regex });
+  }
+
+  /**
+   * The value of the header with the given name in the request must be in a
+   * range of values.
+   *
+   * @param headerName the name of the header to match against
+   * @param start Match on values starting at and including this value
+   * @param end Match on values up to but not including this value
+   */
+  static valuesIsInRange(headerName: string, start: number, end: number): HeaderMatch {
```

Comment:
Can we also add `public` here everywhere in front of the static factory methods?

Reply:
Addressed